bin/registerServerTopology.php

```diff
@@ -68,70 +81,92 @@
 /**
  * Assign options to variables
  */
-try {
-    if (!isset($opt['u'], $opt['t'], $opt['h'], $opt['n'])) {
-        throw new \InvalidArgumentException(
-            PHP_EOL . 'missing parameter: -u -t -h -n are mandatories:' . PHP_EOL . $helpMessage
-        );
+$configOptions = [];
+if (isset($opt['template'])) {
+    $configTemplate = parseTemplateFile($opt['template']);
+    try {
+        $configOptions = setConfigOptionsFromTemplate($configTemplate, $helpMessage);
+    } catch (\InvalidArgumentException $e) {
+        exit($e->getMessage());
     }
+} else {
+    try {
+        if (!isset($opt['u'], $opt['t'], $opt['h'], $opt['n'])) {
+            throw new \InvalidArgumentException(
+                PHP_EOL . 'missing parameter: -u -t -h -n are mandatories:' . PHP_EOL . $helpMessage
+            );
+        }
 
-    $username = $opt['u'];
-    $serverType = in_array(strtolower($opt['t']), SERVER_TYPES)
-        ? strtolower($opt['t'])
-        : false;
-
-    if (!$serverType) {
-        throw new \InvalidArgumentException(
-            "-t must be one of those value"
-            . PHP_EOL . "Poller, Remote, MAP, MBI" . PHP_EOL
-        );
-    }
+        $configOptions['API_USERNAME'] = $opt['u'];
+        $configOptions['SERVER_TYPE'] = in_array(strtolower($opt['t']), SERVER_TYPES)
+            ? strtolower($opt['t'])
+            : false;
 
-    if (isset($opt['dns'])) {
-        $dns = filter_var($opt['dns'], FILTER_VALIDATE_DOMAIN);
-        if (!$dns) {
+        if (!$configOptions['SERVER_TYPE']) {
             throw new \InvalidArgumentException(
-                PHP_EOL . "Bad DNS Format" . PHP_EOL
+                "-t must be one of those value"
+                    . PHP_EOL . "Poller, Remote, MAP, MBI" . PHP_EOL
             );
         }
+
+        $configOptions['ROOT_CENTREON_FOLDER'] = $opt['root'] ?? 'centreon';
+        $configOptions['HOST_ADDRESS'] = $opt['h'];
+        $configOptions['SERVER_NAME'] = $opt['n'];
+
+        if (isset($opt['dns'])) {
+            $configOptions['DNS'] = filter_var($opt['dns'], FILTER_VALIDATE_DOMAIN);
+            if (!$configOptions['DNS']) {
+                throw new \InvalidArgumentException(
+                    PHP_EOL . "Bad DNS Format" . PHP_EOL
+                );
+            }
+        }
+    } catch (\InvalidArgumentException $e) {
+        exit($e->getMessage());
```

Comment:
The return value is called exit status. This value can be used to determine whether a command completed successfully or unsuccessfully. If the command exits successfully, the exit status will be zero, otherwise it will be a nonzero value.

```suggestion
        echo $e->getMessage(); 
        exit(1);
```